modules/datahandling/requestmap_test.go

```diff
@@ -23,11 +27,8 @@ func TestProjectCreateRequest(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	switch newRequest.(type) {
-	default:
-		t.Fatalf("newRequest is the wrong type, expected: *projectCreateRequest, actual: %s\n", reflect.TypeOf(newRequest))
-	case *projectCreateRequest:
-
+	if reflect.TypeOf(newRequest).String() != "*datahandling.projectCreateRequest" {
```

Comment:
yeah, the request map stuff is not part of this feature set, I was just changing the reflection I was using in the tests to be easier to read. I definitely agree that the request constructors should be struct based; however, if we want to clean this up it should be a separate thing